### PR TITLE
Update meta.yaml

### PR DESCRIPTION
**Pull request by pplonski**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Hey @raybellwaves, I tried to split `dtreeviz` package into two packages:
- `dtreeviz` - only required dependencies installed
- `dtreeviz-extended` - installed additional dependencies (xgboost, lighgbm, pyspark)

The biggest problem is with `pyspark`. Not everyone needs this dependency and additionally it breaks installation on Python 3.9, so `dtreeviz` installed with conda won't work on Python 3.9 (discussion on stackoverflow https://stackoverflow.com/questions/58700384/how-to-fix-typeerror-an-integer-is-required-got-type-bytes-error-when-tryin)

### CODE REVIEW

**Comment by pplonski** [comment]
@raybellwaves if you merge the PR then there will be two packages available in the conda-forge:
- `dtreeviz`
- `dtreeviz-extended` (with pyspark, xgboost and lightgbm)

**Comment by raybellwaves** [comment]
Thanks @pplonski! LGTM.

You wanna add yourself to a maintainer?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -11,40 +11,59 @@ source:
...
+      requires:
+        - pip
+  - name: {{ name }}-extended
+    requirements:
+      host:
+        - pip
+        - python >=3.6
+      run:
+        - python >=3.6
+        - {{ pin_subpackage(name, max_pin='x.x.x.x.x.x') }}
```
Comment (raybellwaves) [inline recipe/meta.yaml:50]: Not familiar with this but just looked it up.
https://docs.conda.io/projects/conda-build/en/latest/resources/variants.html#referencing-subpackages
I'm fine if you are happy with using using the max_pin arg

Comment (pplonski) [reply recipe/meta.yaml]: @raybellwaves I've just used the same convention as in airflow recipe from your [comment](https://github.com/parrt/dtreeviz/issues/144#issuecomment-859656476). Now, I've tried with flag `exact=True` I think it should work as well (there is only one version of `dtreeviz` in the build).

**Comment by raybellwaves** [comment]
@conda-forge-admin, please rerender

**Pull request merged**